plugins/c8y_log_plugin/src/main.rs

```diff
@@ -154,6 +126,72 @@ async fn run(
     }
 }
 
+pub async fn process_mqtt_messages(
+    message: Option<Message>,
+    plugin_config: &LogPluginConfig,
+    mqtt_client: &mut Connection,
+    http_client: &mut JwtAuthHttpProxy,
+    config_file: &Path,
+) -> Result<(), anyhow::Error> {
+    let response_topic_health = Topic::new_unchecked("tedge/health/c8y-log-plugin");
+    if let Some(message) = message {
+        if is_c8y_bridge_up(&message) {
+            let plugin_config = read_log_config(config_file);
+            let () = handle_dynamic_log_type_update(&plugin_config, mqtt_client).await?;
+        }
+        match message.topic.name.as_str() {
+            "tedge/health-check" | "tedge/health-check/c8y-log-plugin" => {
+                let health_status = json!({
+                    "status": "up",
+                    "pid": process::id(),
+                    "time": OffsetDateTime::now_utc().unix_timestamp(),
+                })
+                .to_string();
+
+                let health_message = Message::new(&response_topic_health, health_status);
+                let _ = mqtt_client.published.send(health_message).await;
+                return Ok(());
+            }
+            _ => {
+                if let Ok(payload) = message.payload_str() {
+                    let result = match payload.split(',').next().unwrap_or_default() {
+                        "522" => {
+                            info!("Log request received: {payload}");
+                            // retrieve smartrest object from payload
+                            let maybe_smartrest_obj = SmartRestLogRequest::from_smartrest(payload);
+                            if let Ok(smartrest_obj) = maybe_smartrest_obj {
+                                handle_logfile_request_operation(
+                                    &smartrest_obj,
+                                    plugin_config,
+                                    mqtt_client,
+                                    http_client,
+                                )
+                                .await
+                            } else {
+                                error!("Incorrect SmartREST payload: {}", payload);
+                                Ok(())
+                            }
+                        }
+                        _ => {
+                            // Ignore operation messages not meant for this plugin
+                            Ok(())
+                        }
+                    };
+
+                    if let Err(err) = result {
+                        let error_message =
+                            format!("Handling of operation: '{}' failed with {}", payload, err);
+                        error!("{}", error_message);
+                    }
+                }
+                return Ok(());
+            }
+        }
+    }
+    // message is None and the connection has been closed
+    Ok(())
```

Comment:
Having this function called even when there is no messages introduced a difference of behavior. Previously, the loop was ended on connection close. This is no more the case.

Reply:
fixed